spikeinterface/qualitymetrics/misc_metrics.py

```diff
@@ -488,6 +485,207 @@ def compute_amplitudes_cutoff(waveform_extractor, peak_sign='neg',
+                      "Use the `postprocessing.compute_spike_locations()` function. "
+                      "Drift metrics will be set to NaN")
+        empty_dict = {unit_id: np.nan for unit_id in waveform_extractor.unit_ids}
+        if return_positions:
+            return res(empty_dict, empty_dict), np.nan
+        else:
+            return res(empty_dict, empty_dict)
+
+    recording = waveform_extractor.recording
+    sorting = waveform_extractor.sorting
+    unit_ids = waveform_extractor.unit_ids
+    maximum_drift = {}
+    cumulative_drift = {}
+    maximum_drift_segs = {unit_id: [] for unit_id in unit_ids}
+    cumulative_drift_segs = {unit_id: [] for unit_id in unit_ids}
+    interval_samples = int(interval_s * waveform_extractor.sampling_frequency)
+    assert direction in spike_locations.dtype.names, (
+        f"Direction {direction} is invalid. Available directions: "
+        f"{spike_locations.dtype.names}"
+    )
+
+    # if a segment is shorter than 2 * interval_s, we set the drift metrics to -1
+    for segment_index in range(recording.get_num_segments()):
+        seg_length = recording.get_num_samples(segment_index)
+        if  seg_length <= 2 * interval_samples:
+            warnings.warn(f"Segment {segment_index} is too short to compute drift metrics "
+                          f"with interval_s of {interval_s}. It needs to be at least {2*interval_s}s long")
+            for unit_id in unit_ids:
+                maximum_drift_segs[unit_id].append(-1)
+                cumulative_drift_segs[unit_id].append(-1)
+        else:
+            num_bin_edges = seg_length // interval_samples + 1
+            bins = np.arange(num_bin_edges) * interval_samples
+            spike_vector = sorting.to_spike_vector()
+
+            # compute median positions (if less than min_spikes_per_interval, median position is 0)
+            median_positions = np.nan * np.zeros((len(unit_ids), num_bin_edges - 1))
+            for bin_index, (start_frame, end_frame) in enumerate(zip(bins[:-1], bins[1:])):
+                i0 = np.searchsorted(spike_vector['segment_ind'], segment_index)
+                i1 = np.searchsorted(spike_vector['segment_ind'], segment_index + 1)
+                spikes_in_segment = spike_vector[i0:i1]
+                spike_locations_in_segment = spike_locations[i0:i1]
+                i0 = np.searchsorted(spikes_in_segment['sample_ind'], start_frame)
+                i1 = np.searchsorted(spikes_in_segment['sample_ind'], end_frame)
+                spikes_in_bin = spikes_in_segment[i0:i1]
+                spike_locations_in_bin = spike_locations_in_segment[i0:i1][direction]
+
+                for unit_ind in np.arange(len(unit_ids)):
+                    mask = spikes_in_bin['unit_ind'] == unit_ind
+                    if np.sum(mask) >= min_spikes_per_interval:
+                        median_positions[unit_ind, bin_index] = np.median(spike_locations_in_bin[mask])
```

Comment:
Should we add a "direction" option to compute only in one direction ?

Reply:
we do have `direction='y'`

Reply:
see https://github.com/alejoe91/spikeinterface/blob/missing-metrics/spikeinterface/qualitymetrics/misc_metrics.py#L638